### DISCUSSION TITLE
Revalidating Server Components when navigating on the client

### DISCUSSION THREAD

**Discussion by antonmoiseev** [vercel/next.js#44699] [App Router]
Is it possible to revalidate Server Components when navigating to the page on the client using Link component or useRouter()? Currently "no-store" Server Components are revalidated only if the entire page is refreshed.

**Comment by logemann** [comment]
Yeah. Official comment would be nice. Doesn't make sense in my view to not do revalidation on next/link.

**Comment by karlhorky** [comment]
It sounds like you're having the issue with soft navigation that a number of community members are experiencing.

I would suggest that you add your feedback to the thread over here:

- #42991
